src/Value/Color.php

```diff
@@ -56,12 +56,19 @@ public static function parse(ParserState $oParserState, $bIgnoreCase = false)
                         $oParserState->currentLine()
                     ),
                 ];
-            } else {
+            } elseif ($oParserState->strlen($sValue) === 6) {
                 $aColor = [
                     'r' => new Size(intval($sValue[0] . $sValue[1], 16), null, true, $oParserState->currentLine()),
                     'g' => new Size(intval($sValue[2] . $sValue[3], 16), null, true, $oParserState->currentLine()),
                     'b' => new Size(intval($sValue[4] . $sValue[5], 16), null, true, $oParserState->currentLine()),
                 ];
+            } else {
+                // the size of input color string is incorrect, use black color as fallback
+                $aColor = [
+                    'r' => new Size(0, null, true, $oParserState->currentLine()),
+                    'g' => new Size(0, null, true, $oParserState->currentLine()),
+                    'b' => new Size(0, null, true, $oParserState->currentLine()),
+                ];
```

Comment:
If the value is not valid, we should throw an `UnexpectedTokenException`.  This wil be caught by `RuleSet::parseRuleSet()`, resulting in the invalid property declaration being dropped (while other valid properties are retained, and which matches the behaviour of most browsers).

```suggestion
                throw new UnexpectedTokenException(
                    'Invalid hex color value', 
                    $sValue, 
                    'custom', 
                    $oParserState->currentLine()
                );
```

Reply:
I agree.

Reply:
Thanks, yes this is a much better solution.